raco/myrial/interpreter.py

```diff
@@ -241,28 +243,29 @@ def bagcomp(self, from_clause, where_clause, emit_clause,
 
         if any(raco.expression.expression_contains_aggregate(ex)
                for name, ex in emit_args):
-                    op = groupby.groupby(op, emit_args, implicit_group_by_cols,
-                                         statemods)
+            op = groupby.groupby(op, emit_args, implicit_group_by_cols,
+                                 statemods)
         else:
             if statemods:
                 return raco.algebra.StatefulApply(emit_args, statemods, op)
-            if (len(from_args) == 1 and len(emit_clause) == 1 and
-                isinstance(emit_clause[0],
-                           (TableWildcardEmitArg, FullWildcardEmitArg))):
-                op = op
-            op = raco.algebra.Apply(emit_args, op)
+            if not (len(from_args) == 1 and len(emit_clause) == 1 and
+                    isinstance(emit_clause[0],
+                               (TableWildcardEmitArg, FullWildcardEmitArg))):
+                op = raco.algebra.Apply(emit_args, op)
 
         if orderby_clause:
             if limit_clause is None:
-                raise Exception("Order By queries must use the Limit operator")
+                raise Exception(
```

Comment:
Let's not raise a generic `Exception` here; I think the closest semantic match might be `InvalidStatementException`.